### PR TITLE
location entity, more functionality added

### PR DESCRIPTION
**Pull request by tstrych**
It's not done yet. You can find some TODO's there.
But please give me a feedback @oshtaier or @abalakh  on what is already written.

### CODE REVIEW

File: tests/foreman/ui_airgun/test_location.py [hunk 1]
```diff
@@ -27,3 +37,338 @@ def test_positive_create(session):
...
+    loc_name = gen_string('alpha')
+    with session:
+        session.location.create({'name': loc_name})
+        session.location.delete(loc_name)
+        assert not session.location.search(loc_name)
+
+
+@run_only_on('sat')
+@tier2
+def test_positive_create_with_location_and_org(session):
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_location.py]: remove that test from both old ui and here as it doesn't make sense. We verify that functionality across application when select both org and loc

File: tests/foreman/ui_airgun/test_location.py [hunk 2]
```diff
@@ -27,3 +37,338 @@ def test_positive_create(session):
...
+    subnet = entities.Subnet(
+        network=ip_addres,
+        mask='255.255.255.0',
+    ).create()
+    loc = entities.Location().create()
+    with session:
+        session.location.update(
+            loc.name, {'subnets.resources.assigned': [subnet.name]})
+        loc_values = session.location.read(loc.name)
+        assert loc_values['subnets']['resources']['assigned'][0] == \
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_location.py]: probably move `subnet.name + " (" + ip_addres + "/24)"` outside in variable
`subnet_name = "{}({}/24)".format(subnet.name, ip_address)`
or even
`subnet_name = "{}({}/{})".format(subnet.name, ip_address, subnet.cidr)`

**Comment by tstrych** [comment]
@abalakh or @oshtaier please give me a feedback

File: tests/foreman/ui_airgun/test_location.py [hunk 3]
```diff
@@ -27,3 +37,351 @@ def test_positive_create(session):
...
+    :BZ: 1479736
+
+    :CaseLevel: Integration
+    """
+    user = entities.User().create()
+    loc = entities.Location().create()
+    with session:
+        session.organization.select(org_name='Any Organization')
+        session.location.select(loc_name=loc.name)
+        session.location.update(
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_location.py]:
```
+        session.location.update(
+            loc.name, {'users.resources.assigned': [user.login]})
+        loc_values = session.location.read(loc.name)
+        user_values = session.user.read(user.login)
+        assert loc_values['users']['resources']['assigned'][0] == user.login
+        assert user_values['locations']['resources']['assigned'][0] == loc.name
```
that is redundant and goes against initial goal of test case

File: tests/foreman/ui_airgun/test_location.py [hunk 4]
```diff
@@ -27,3 +37,351 @@ def test_positive_create(session):
...
+
+def test_positive_delete(session):
+    loc_name = gen_string('alpha')
+    with session:
+        session.location.create({'name': loc_name})
+        session.location.delete(loc_name)
+        assert not session.location.search(loc_name)
+
+
+@run_only_on('sat')
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_location.py]: remove `@run_only_on('sat')` from all test cases

File: tests/foreman/ui_airgun/test_location.py (hunk 3 above)
Comment (oshtaier) [reply tests/foreman/ui_airgun/test_location.py]: oops, that is different bug, nvm

File: tests/foreman/ui_airgun/test_location.py [hunk 5]
```diff
@@ -27,3 +37,351 @@ def test_positive_create(session):
...
+        user_values = session.user.read(user.login)
+        assert loc.name in user_values['locations']['resources']['assigned']
+        session.location.update(loc.name, {'users.all_users': False})
+        user_values = session.user.read(user.login)
+        assert loc.name in user_values['locations']['resources']['unassigned']
+
+
+@run_only_on('sat')
+@tier2
+def test_positive_update_hostgroup(session):
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_location.py]: not sure why did you decide to test functionality in reverse order as it doesn't actually test remove functionality

File: tests/foreman/ui_airgun/test_location.py [hunk 6]
```diff
@@ -27,3 +37,351 @@ def test_positive_create(session):
...
+        session.location.update(
+            loc.name, {'media.resources.unassigned': [media.name]})
+        loc_values = session.location.read(loc.name)
+        assert len(loc_values['media']['resources']['assigned']) == 0
+        assert media.name in loc_values['media']['resources']['unassigned']
+
+
+@run_only_on('sat')
+@tier2
+def test_positive_update_template(session):
```
Comment (oshtaier) [inline tests/foreman/ui_airgun/test_location.py]: same here

File: tests/foreman/ui_airgun/test_location.py [hunk 7]
```diff
@@ -27,3 +37,351 @@ def test_positive_create(session):
...
+    :expectedresults: organization is added to location
+
+    :CaseLevel: Integration
+    """
+    org = entities.Organization().create()
+    loc = entities.Location().create()
+    with session:
+        session.location.update(
+            loc.name, {'organizations.resources.assigned': [org.name]})
+
```
Comment (abalakh) [inline tests/foreman/ui_airgun/test_location.py]: nitpick: redundant line

File: tests/foreman/ui_airgun/test_location.py (hunk 5 above)
Comment (oshtaier) [reply tests/foreman/ui_airgun/test_location.py]: That means
```
         session.location.update(
             loc.name,
             {'host_groups.all_hostgroups': False,
              'host_groups.resources.unassigned': [hostgroup.name]}
         )
 
         loc_values = session.location.read(loc.name)
         assert loc_values[
             'host_groups']['resources']['unassigned'][0] == hostgroup.name
```
```
         assert len(loc_values[
             'host_groups']['resources']['unassigned']) == 1
```
?

**Comment by oshtaier** [comment]
ACK pending test run after all the changes
I checked pages and you did everything good, so will merge right after results provided

**Comment by tstrych** [comment]
I'm sorry I write you the comment but the comment was rewritten after my forced push, 
I don't know if you receive it, but I write there, by default the hostgroup and template have checked all_values, so is much more easier firstly removed them and then add back. 
Ok, I will provide tests :)

**Comment by tstrych** [comment]
@oshtaier  here it is.
```
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.5, pytest-3.4.0, py-1.5.3, pluggy-0.6.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/tstrych/Documents/robottelo, inifile:
plugins: wait-for-1.0.9, services-1.2.1, mock-1.6.3
collected 14 items                                                                                                                                                                                                
2018-07-09 15:17:17 - conftest - DEBUG - Collected 14 test cases

2018-07-09 15:17:17 - conftest - DEBUG - Deselected test tests.foreman.ui_airgun.test_location.test_positive_update_with_all_users

2018-07-09 15:17:17 - conftest - DEBUG - Deselected test tests.foreman.ui_airgun.test_location.test_positive_update_with_all_users_setting_only


tests/foreman/ui_airgun/test_location.py .........s..                                                                                                                                                       [100%]

=============================================================================================== 2 tests deselected ================================================================================================
============================================================================== 11 passed, 1 skipped, 2 deselected in 665.29 seconds ===============================================================================
```

**Pull request merged**